### PR TITLE
chore(react-scheduler): add Scheduler Appointment Form

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**

### CODE REVIEW

File: packages/dx-react-scheduler-material-ui/src/plugins/appointment-form.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { withComponents } from '@devexpress/dx-react-core';
+import { AppointmentForm as AppointmentFormBase } from '@devexpress/dx-react-scheduler';
+import { Popup } from '../templates/appointment-form/popup';
+import { CommandButton } from '../templates/appointment-form/command-button';
+import { TextEditor } from '../templates/appointment-form/text-editor';
+import { DateEditor } from '../templates/appointment-form/date-editor';
+import { AllDayEditor } from '../templates/appointment-form/all-day-editor';
+import { PopupContainer } from '../templates/appointment-form/popup-container';
```
Comment (SergeyAlexeev) [inline packages/dx-react-scheduler-material-ui/src/plugins/appointment-form.jsx]: PopupContainer -> Container

File: packages/dx-react-scheduler-material-ui/src/templates/appointment-form/all-day-editor.test.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { createShallow } from '@material-ui/core/test-utils';
+import { AllDayEditor } from './all-day-editor';
+
+describe('Appointment Form', () => {
+  let shallow;
+  beforeAll(() => {
+    shallow = createShallow({ dive: true });
+  });
+  describe('AllDayEditor', () => {
+    it('should pass rest props to the root element', () => {
```
Comment (SergeyAlexeev) [inline packages/dx-react-scheduler-material-ui/src/templates/appointment-form/all-day-editor.test.jsx]: The test name is 'should pass rest props to the root element' but you check the 'className' prop.

File: packages/dx-react-scheduler-material-ui/src/plugins/appointment-form.jsx (hunk 1 above)
Comment (MaximKudriavtsev) [reply packages/dx-react-scheduler-material-ui/src/plugins/appointment-form.jsx]: Or PopupContent?

File: packages/dx-react-scheduler-material-ui/src/templates/appointment-form/all-day-editor.test.jsx (hunk 2 above)
Comment (MaximKudriavtsev) [reply packages/dx-react-scheduler-material-ui/src/templates/appointment-form/all-day-editor.test.jsx]: Yeah, this component has not a `className` property. In this case, `className` will be spread with `restProps`.

File: packages/dx-react-scheduler-material-ui/src/plugins/appointment-form.jsx (hunk 1 above)
Comment (SergeyAlexeev) [reply packages/dx-react-scheduler-material-ui/src/plugins/appointment-form.jsx]: I'd prefer 'Container')

File: packages/dx-react-scheduler-material-ui/src/templates/appointment-form/all-day-editor.test.jsx (hunk 2 above)
Comment (SergeyAlexeev) [reply packages/dx-react-scheduler-material-ui/src/templates/appointment-form/all-day-editor.test.jsx]: I see. OK, let's keep it as is.

File: packages/dx-react-scheduler/src/plugins/appointment-form.jsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+} from '@devexpress/dx-react-core';
+import {
+  setAppointment,
+  COMMIT_COMMAND_BUTTON,
+  CANCEL_COMMAND_BUTTON,
+} from '@devexpress/dx-scheduler-core';
+
+const defaultMessages = {
+  allDayText: 'All Day',
+  titleLabel: 'Subject',
```
Comment (SergeyAlexeev) [inline packages/dx-react-scheduler/src/plugins/appointment-form.jsx]: We use 'Subject' as a default value for a title)) It looks a bit strange. Can we change it to `titleLabel: 'Title'` ?

Comment (MaximKudriavtsev) [reply packages/dx-react-scheduler/src/plugins/appointment-form.jsx]: I'm not against)

File: packages/dx-react-scheduler/src/plugins/appointment-form.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      <Plugin
+        name="AppointmentForm"
+      >
+        <Action name="toggleFormVisibility" action={this.toggleVisibility} />
+        <Action name="setFormAppointment" action={this.setAppointment} />
+
+        <Popup
+          visible={visible}
+        >
+          <PopupContainer>
```
Comment (SergeyAlexeev) [inline packages/dx-react-scheduler/src/plugins/appointment-form.jsx]: PopupContainer is still here)

**Pull request merged**